addOns/network/src/test/java/org/zaproxy/addon/network/internal/client/HttpSenderImplUnitTest.java

```diff
@@ -958,6 +958,46 @@ void shouldNotContainUserObjectWithConnectionClosedIfNotClosed(
             assertThat(message.getUserObject(), is(nullValue()));
         }
 
+        @Test
+        @Timeout(10)
+        void shouldNotRetry429WithRetryAfter() throws Exception {
```

Comment:
This should be moved under `Request` and be like:
```suggestion
        @Timeout(10)
        @ParameterizedTest
        @MethodSource(
                "org.zaproxy.addon.network.internal.client.HttpSenderImplUnitTest#statusCodesForRetryAndSendAndReceiveMethods")
        void shouldNotBeRetriedOnAnyStatusCode(int statusCode, SenderMethod method)
                throws Exception {
            // Given
            server.setHttpMessageHandler(
                    (ctx, msg) -> {
                        msg.setResponseHeader(
                                "HTTP/1.1 "
                                        + statusCode
                                        + "\r\nretry-after: 3600\r\ncontent-length: 13\r\n\r\n");
                        msg.getResponseBody().setBody("Response Body");
                    });
            // When
            httpSender.sendAndReceive(message);
            // Then
            assertThat(server.getReceivedMessages(), hasSize(1));
        }
```
with `statusCodesForRetryAndSendAndReceiveMethods` as:
```java
    static Stream<Arguments> statusCodesForRetryAndSendAndReceiveMethods() {
        var statusCodes = new ArrayList<Integer>();
        for (int i = 200; i < 600; i++) {
            statusCodes.add(i);
        }
        return statusCodes.stream()
                .flatMap(
                        statusCode -> sendAndReceiveMethods().map(sm -> arguments(statusCode, sm)));
    }
```

Reply:
Fixed. Thanks for the code!